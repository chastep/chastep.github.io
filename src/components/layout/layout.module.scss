@use "src/assets/styles/variables";

.layout {
  display: grid;
  margin: 0 auto;
  gap: variables.$layout-gap;
  max-width: variables.$layout-width;
  grid-template-columns: repeat(12, 1fr);
}
