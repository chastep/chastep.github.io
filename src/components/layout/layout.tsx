import React, { type FC, type ReactNode, useEffect } from "react";

import { useTheme } from "@/hooks/use-theme";

import * as styles from "./layout.module.scss";

interface LayoutProps {
  children: ReactNode;
}

const Layout: FC<LayoutProps> = ({ children }) => {
  const [{ mode }] = useTheme();

  useEffect(() => {
    document.documentElement.className = mode;
  }, [mode]);

  return <div className={styles.layout}>{children}</div>;
};

export { Layout };
