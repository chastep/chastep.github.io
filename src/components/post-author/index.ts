export { PostAuthor } from "./post-author";
