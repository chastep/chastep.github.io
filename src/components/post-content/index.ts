export { PostContent } from "./post-content";
