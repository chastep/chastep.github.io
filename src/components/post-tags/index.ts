export { PostTags } from "./post-tags";