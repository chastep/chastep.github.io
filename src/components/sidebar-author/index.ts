export { SidebarAuthor } from "./sidebar-author";