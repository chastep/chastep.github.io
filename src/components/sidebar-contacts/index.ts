export { SidebarContacts } from "./sidebar-contacts";
