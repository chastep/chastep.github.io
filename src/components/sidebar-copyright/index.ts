export { SidebarCopyright } from "./sidebar-copyright";
