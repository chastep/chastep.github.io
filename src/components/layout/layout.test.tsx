import React from "react";
import { test, describe, expect } from "bun:test";

import { Layout } from "@/components/layout";
import { createSnapshotsRenderer, renderWithCoilProvider } from "@/utils/render-with-coil-provider";

describe("Layout", () => {
  const LayoutWithChildren = () => <Layout>test</Layout>;

  test("renders correctly", () => {
    const tree = createSnapshotsRenderer(<LayoutWithChildren />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("dark theme is set correctly", () => {
    window.localStorage.setItem(
      "diesel:theme-atom",
      JSON.stringify({ mode: "dark" }),
    );

    renderWithCoilProvider(<LayoutWithChildren />);

    window.localStorage.removeItem("diesel:theme-atom");

    expect(document.documentElement.className).toBe("dark");
  });

  test("light theme is set correctly", () => {
    renderWithCoilProvider(<LayoutWithChildren />);

    expect(document.documentElement.className).toBe("light");
  });
});
