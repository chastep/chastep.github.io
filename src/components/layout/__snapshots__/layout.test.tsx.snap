// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
{
  "$$typeof": Symbol(react.test.json),
  "children": [
    "test",
  ],
  "props": {
    "className": undefined,
  },
  "type": "div",
}
`;
